### PR TITLE
(commit): moved cache (#240)

### LINKED ISSUE #240: Configure cache
Currently when using commitizen it dumps the commitizen.json file in the home directory (as described [here](https://github.com/commitizen/cz-cli#retrying-failed-commits)).
In Windows this doesn't exactly form a problem (you rarely see your homefolder), but when using Linux you're dropped right into the home folder when opening a file manager.
I'm no real fan of cluttering my home directory with config files and end up deleting the .json file every time, but this is rather cumbersome to do every time I open a file manager.

My suggestion is to allow configuring the path where the commitizen.json is stored or rename it to .comittizen.json or something so that it's hidden.

as you can see below, it really sticks out like a sore thumb

### PR DESCRIPTION
**Pull request by dealloc**
Moved the cache from the home folder to a designated config folder

This fixes #240 by moving the commitizen.json cache file out of the home folder

### CODE REVIEW

File: src/commitizen/commit.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import path from 'path';
-import homeOrTmp from 'home-or-tmp';
+import ConfigDir from 'application-config-path';
```
Comment (LinusU) [inline src/commitizen/commit.js]: I think that the convention is that initial uppercase letter is for classes, can you change this name to `configDir` instead? thanks

Comment (dealloc) [inline src/commitizen/commit.js]: Certainly, hang on :)

**Comment by dealloc** [comment]
@LinusU I don't see why the builds should fail, could this be related to #205 ?

**Comment by LinusU** [comment]
👍 I'll look into the failing build when I'm not at work

**Comment by dealloc** [comment]
I'm pulling my hairs out over this one, I literally replaced a string with another string ?_?

**Comment by jimthedev** [comment]
I could be wrong, but I am pretty sure this is failing simply because if the directory doesn't exist, it isn't being created first. Your PR should probably create that directory before trying to create the file.

**Comment by dealloc** [comment]
I'm going to give that a try

**Comment by LinusU** [comment]
@dealloc Maybe it's better to use this package since it's actually meant for cache? The current one is meant for config which isn't really what this is...

https://www.npmjs.com/package/cachedir

**Comment by dealloc** [comment]
does this also handle creating the directories etc? or would I need to handle that myself?

**Comment by LinusU** [comment]
This works the same, it will just give back the path.

Ahh, you're right, we should probably fix that first. PR welcome if you have the time, otherwise I'll try and fix it as soon as possible.

**Comment by dealloc** [comment]
I suggest for now we could just use the package I already included and just update it when the other one is fixed. The main thing right now is that it won't commit and I have no clue why (I suspect #205 is the culprit here).

**Pull request closed without merging**